src/components/AuthForm/SignInForm/index.tsx

```diff
@@ -1,47 +1,63 @@
-import React from 'react'
+import React, { useState } from 'react'
 import styles from './SignInForm.module.scss'
 import Button from '@/UI/Button'
-import { useForm } from 'react-hook-form'
+import { SubmitHandler, useForm } from 'react-hook-form'
 import AuthInput from '../AuthInput'
 import AuthErrorMessage from '../AuthErrorMessage'
 import { useTranslation } from 'next-i18next'
+import { signIn } from '@/firebase/clientApp'
+import { useRouter } from 'next/router'
+
+export interface FormInValues {
+  email: string
+  password: string
+}
 
 const SignInForm = () => {
-  const onSubmit = (data: object) => {
-    console.log(JSON.stringify(data))
-    reset()
+  const router = useRouter()
+  const [signinError, setSigninError] = useState<string>('')
+
+  const onSubmit: SubmitHandler<FormInValues> = async (data) => {
+    const { email, password } = data
+
+    const { error } = await signIn(email, password)
+
+    if (error) {
+      setSigninError('Неверный пароль')
+      return
+    } else {
+      reset()
+      router.push('/editor')
+    }
   }
 
   const {
     handleSubmit,
     reset,
     control,
     formState: { errors, isValid },
-  } = useForm({
+  } = useForm<FormInValues>({
     mode: 'onChange',
   })
 
   const { t } = useTranslation('auth')
 
   return (
     <form action="#" className={styles.container} onSubmit={handleSubmit(onSubmit)}>
-      <AuthInput
+      <AuthInput<FormInValues>
         control={control}
-        name="login"
-        type="text"
+        name="email"
+        type="email"
         rules={{
-          required: 'nologin',
+          required: 'noemail',
           pattern: {
-            value: /^(?=.*[A-Za-z0-9]$)[A-Za-z][A-Za-z\d.-]{0,19}$/,
-            message: 'wrongInput',
-          },
-          minLength: {
-            value: 5,
-            message: 'min',
+            value:
+              /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```

Comment:
try to use this one `/^[^\s@]+@[^\s@]+.[^\s@]+$/`

Reply:
fixed in the last commit